engine/test/unit/dataobjects/__init__.py

```diff
@@ -14,8 +15,8 @@ class MockNested:
 @dataclass
```

Comment:
Leo in this class are the two ways you can add the metadata: I put it in to test but it can result in a complication as someone can put something other than metadata from the json_schema_extra and that will break the json_schema. What are your thoughts on this?